src/ophyd_async/epics/demo/__init__.py

```diff
@@ -56,8 +56,9 @@ def __init__(self, prefix: str, name: str = "", sensor_count: int = 3) -> None:
         )
 
         # Makes read() produce the values of all sensors
-        self.set_readable_signals(
-            read=[sensor.value for sensor in self.sensors.values()],
+        self.add_readables(
```

Comment:
Please can you remove this section and add the context manager around the definition of `self.sensors`
```python
with self.add_children_as_readable():
    self.sensors = ...
```

Reply:
Done in d72889a995bd8bfa579db8fe1df69be14d75a626